crates/next-core/src/app_source.rs

```diff
@@ -289,20 +292,18 @@ async fn create_app_source_for_directory(
     let mut layouts = layouts;
     let mut sources = Vec::new();
     let mut page = None;
-    let mut layout = None;
+    let mut files = HashMap::new();
     if let DirectoryContent::Entries(entries) = &*input_dir.read_dir().await? {
         for (name, entry) in entries.iter() {
-            if let DirectoryEntry::File(file) = entry {
+            if let &DirectoryEntry::File(file) = entry {
                 if let Some((name, _)) = name.rsplit_once('.') {
                     match name {
                         "page" => {
                             page = Some(file);
                         }
-                        "layout" => {
-                            layout = Some(file);
+                        "layout" | "error" | "loading" | "template" | "not-found" | "head" => {
+                            files.insert(name.to_string(), file);
```

Comment:
Could we have a `FileType` enum with `TryFrom<String>`/`Into<String>` impls instead?

`files` could either be a `HashMap<FileType, file>` or a `FileTypeMap` which has specific fields for `layout`, `error`, etc. and an `Iterator<Item = (FileType, File)>` impl. I think going for the more specific `FileTypeMap` is better here.